karabo_data/cli/make_virtual_cxi.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import logging
+import os
+import os.path as osp
+import re
+import sys
+
+from karabo_data import RunDirectory
+from karabo_data.components import AGIPD1M, LPD1M
+from karabo_data.exceptions import SourceNameError
+
+
+def _get_detector(data, min_modules):
+    for cls in (AGIPD1M, LPD1M):
+        try:
+            return cls(data, min_modules=min_modules)
+        except SourceNameError:
+            continue
+
+
+def main(argv=None):
+    ap = argparse.ArgumentParser('karabo-data-make-virtual-cxi')
+    ap.add_argument('run_dir', help="Path to an EuXFEL run directory")
+    # Specifying a proposal directory & a run number is the older interface.
+    # If the run_number argument is passed, run_dir is used as proposal.
+    ap.add_argument('run_number', nargs="?", help=argparse.SUPPRESS)
+    ap.add_argument(
+        '-o', '--output',
+        help="Filename or path for the CXI output file. "
+             "By default, it is written in the proposal's scratch directory."
+    )
+    ap.add_argument(
+        '--min-modules', type=int, default=9, metavar='N',
+        help="Include trains where at least N modules have data (default 9)"
+    )
+    args = ap.parse_args(argv)
+    out_file = args.output
+
+    logging.basicConfig(level=logging.INFO)
+
+    if args.run_number:
+        # proposal directory, run number
+        run  = 'r%04d' % int(args.run_number)
+        proposal = args.run_dir
+        run_dir = osp.join(args.run_dir, 'proc', run)
+        if out_file is None:
+            out_file = osp.join(proposal, 'scratch', '{}_detectors_virt.cxi'.format(run))
+
+    else:
+        # run directory
+        run_dir = os.path.abspath(args.run_dir)
+        if out_file is None:
+            m = re.search(r'/proc/(r\d{4})/?$', run_dir)
```

Comment:
No particular reason, it was just copied from hdf5-virtualise, which doesn't handle raw data.

I'll add that, although it will mean that the default names produced change to look like `r0123_proc_detectors_virt.cxi`.